### PR TITLE
fix: update VLM first-microbatch state during gradient accumulation

### PR DESCRIPTION
**Pull request by HuiyingLi**
## Summary

- call `prepare_after_first_microbatch()` after the first VLM finetune microbatch
- apply the same first-microbatch state transition to VLM KD
- add unit coverage for the first-microbatch transition in VLM finetune and KD

## Why

`prepare_for_grad_accumulation()` marks the next pass as the first microbatch, but the VLM recipes did not clear that state after the first forward/backward when gradient accumulation was greater than one. In PP + GA runs this left the first-microbatch state active for later microbatches, which can interact badly with FP8/FSDP preparation state.

## Validation

- `pytest -q tests/unit_tests/recipes/test_finetune_vlm_helpers.py tests/unit_tests/recipes/test_vlm_kd_distributed.py`
- `ruff check nemo_automodel/recipes/vlm/finetune.py nemo_automodel/recipes/vlm/kd.py tests/unit_tests/recipes/test_finetune_vlm_helpers.py tests/unit_tests/recipes/test_vlm_kd_distributed.py`
- `python -m compileall -q nemo_automodel/recipes/vlm/finetune.py nemo_automodel/recipes/vlm/kd.py tests/unit_tests/recipes/test_finetune_vlm_helpers.py tests/unit_tests/recipes/test_vlm_kd_distributed.py`
- `git diff --check`

### CODE REVIEW

**Review by akoumpa — APPROVED** [review]
Thanks @HuiyingLi , 

I will make a ticket to follow on some API redesign this PR surfaced.
